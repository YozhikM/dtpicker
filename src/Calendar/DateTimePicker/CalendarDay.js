/* @flow */

import React from 'react';

type Props = {|
  day: number,
  onClick?: (day: number) => void,
  style?: any,
|};

export default class CalendarDay extends React.Component<Props, void> {
  onClick = () => {
    const { onClick, day } = this.props;

    if (onClick) {
      onClick(day);
    }
  };

  render() {
    const { day, style } = this.props;

    return (
      <td style={style} onClick={this.onClick}>
        {day}
      </td>
    );
  }
}
