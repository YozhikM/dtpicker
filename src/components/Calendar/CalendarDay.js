/* @flow */

import React from 'react';
import { getDate } from 'date-fns';

type Props = {|
  date: Date,
  onClick?: (date: Date) => void,
  style: any
|};

type State = void;

class CalendarDay extends React.Component<Props, State> {
  onClick = () => {
    const { onClick, date } = this.props;
    if (onClick) {
      onClick(date);
    }
  };

  render() {
    const { date, style } = this.props;
    let dayOfMonth = getDate(date);
    return <td style={style} onClick={this.onClick}>{dayOfMonth}</td>;
  }
}

export default CalendarDay;
