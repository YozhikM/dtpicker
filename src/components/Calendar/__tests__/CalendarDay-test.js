/* @flow */

import React from 'react';
import CalendarDay from '../CalendarDay';

describe('CalendarDay', () => {
  describe('render()', () => {
    it('basic', () => {
      expect(shallow(<CalendarDay date={new Date(2009, 0, 9)} />)).toMatchSnapshot();
    });

    it('if isActive', () => {
      expect(shallow(<CalendarDay date={new Date(2049, 9, 6)} isActive />)).toMatchSnapshot();
    });

    it('if inRange', () => {
      const date = new Date(2049, 9, 6);
      const startDate = new Date(2049, 4, 5);
      const endDate = new Date(2050, 5, 5);
      expect(
        shallow(<CalendarDay date={date} startDate={startDate} endDate={endDate} />)
      ).toMatchSnapshot();
    });
  });

  it('filterRangeDay()', () => {
    const date = new Date();
    const startDate = new Date(2010, 8, 9);
    const endDate = new Date(2009, 9, 8);
    const wrapper = shallow(<CalendarDay date={date} startDate={startDate} endDate={endDate} />);
    const falsy = wrapper.instance().filterRangeDay(date, startDate, endDate);
    const truthy = wrapper.instance().filterRangeDay(date, date, date);
    expect(falsy).toBeFalsy();
    expect(truthy).toBeTruthy();
  });

  describe('user interactions', () => {
    it('onClick()', () => {
      const spy = jest.fn();
      let date = new Date(2015, 4, 5);
      const wrapper = shallow(<CalendarDay date={date} onClick={spy} />);
      wrapper.find('td').simulate('click');
      expect(spy).toBeCalledWith(date);
      const callbackValue = spy.mock.calls[0][0];
      expect(callbackValue).toEqual(date);
    });
  });
});
