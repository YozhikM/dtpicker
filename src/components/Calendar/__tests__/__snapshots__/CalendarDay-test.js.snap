// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDay render() basic 1`] = `
<td
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": null,
      "border": "1px solid #e4e7e7",
      "color": "#34495e",
    }
  }
>
  9
</td>
`;

exports[`CalendarDay render() if inRange 1`] = `
<td
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#9bb0c5",
      "border": "1px solid transparent",
      "color": "white",
    }
  }
>
  6
</td>
`;

exports[`CalendarDay render() if isActive 1`] = `
<td
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#34495e",
      "border": "1px solid transparent",
      "color": "white",
    }
  }
>
  6
</td>
`;
